src/agents_gbtree.erl

```diff
@@ -90,12 +73,35 @@ group_by(List) ->
                           dict:append_list(K, V, D)
                   end, dict:new(), List)).
 
--spec group_by_colour([[agent()]], pos_integer()) -> [[agent()]].
-group_by_colour(Tiles, N) ->
-    N = 2, % dividing in stripes
-    EveryNth = fun (Rest) ->
-                       [A || {I, A} <- lists:zip(lists:seq(1, length(Tiles)),
-                                                 Tiles),
-                             I rem N == Rest]
-               end,
-    lists:map(EveryNth, [I rem N || I <- lists:seq(1, N)]).
+-spec get_list(agents()) ->
+                      [agent()].
+get_list(Agents) ->
+    gb_trees:values(Agents).
+
+-spec get_tiles(pos_integer(), environment()) -> [{dimension(), tile()}].
+get_tiles(Dist, Env) ->
+    W = (Env#env.world)#world.w,
+    Zeros = [{I, []} || I <- lists:seq(1, W, Dist)],
+    AssignTileToAgent = fun(A = #agent{pos={X, _, _}}) ->
+                                ITile = trunc((X-1)/Dist)*Dist+1,
+                                {ITile, [A]}
+                        end,
+    TiledAgents = lists:map(AssignTileToAgent, get_list(Env#env.agents)),
+    TagTiles = group_by(TiledAgents ++ Zeros),
+    TagTiles.
+
+-spec update_tiles({tile(), environment()}, environment(), config()) -> environment().
```

Comment:
Line 93 is too long: -spec update_tiles({tile(), environment()}, environment(), config()) -> environment()..